gw_spaceheat/actors/home_alone.py

```diff
@@ -277,156 +300,140 @@ def trigger_house_cold_onpeak_event(self) -> None:
-            previous_state = self.state
-
-            if self.is_onpeak():
-                self.storage_declared_ready = False
-                self.full_storage_energy = None
-            time_now = datetime.now(self.timezone) 
-            if (((time_now.hour==6 or time_now.hour==16) and time_now.minute>57)
-                or self.zone_setpoints == {}):
-                self.get_zone_setpoints()
-            
-            # Require weather to move out of initializing state
-            self.get_latest_temperatures()
+        previous_state = self.state
 
-            if not (self.weather and self.temperatures_available):
-                ...
-                # if its been too long kick us into a higher level backup state like ScadaBlind
-            else:
-                if self.state==HomeAloneState.Initializing:
-                    if self.temperatures_available:
-                        if self.is_onpeak():
-                            if self.is_buffer_empty():
-                                if self.is_storage_colder_than_buffer():
-                                    self.trigger_normal_event(HomeAloneEvent.OnPeakBufferFull)
-                                else:
-                                    self.trigger_normal_event(HomeAloneEvent.OnPeakBufferEmpty)
-                            else:
-                                self.trigger_normal_event(HomeAloneEvent.OnPeakBufferFull)
-                        else:
-                            if self.is_buffer_empty():
-                                self.trigger_normal_event(HomeAloneEvent.OffPeakBufferEmpty)
-                            else:
-                                if self.is_storage_ready():
-                                    self.trigger_normal_event(HomeAloneEvent.OffPeakBufferFullStorageReady)
-                                else:
-                                    self.trigger_normal_event(HomeAloneEvent.OffPeakBufferFullStorageNotReady)
+        if self.is_onpeak():
+            self.storage_declared_ready = False
+            self.full_storage_energy = None
 
-                elif self.state==HomeAloneState.HpOnStoreOff:
+        time_now = datetime.now(self.timezone)
+        if ((time_now.hour==6 or time_now.hour==16) and time_now.minute>57) or self.zone_setpoints=={}:
+            self.get_zone_setpoints()
+        
+        self.get_latest_temperatures()
+
+        if not (self.weather and self.temperatures_available):
+            if self.time_since_blind is None:
+                self.time_since_blind = time.time()
+            elif time.time() - self.time_since_blind > 5*60 and self.top_state==HomeAloneTopState.Normal:
+                self.log("Scada is missing weather and/or critical temperatures since at least 5 min.")
+                self.log("Moving into ScadaBlind top state")
+                self.trigger_missing_data()
+        else:
+            if self.time_since_blind is not None:
+                self.time_since_blind = None
```

Comment:
move time_since_blind checks etc up to main - this is logic re top state changes. This loop should only know about normal mode.